### ISSUE TITLE
[RFE] Improve efficiency of diff_string_lists (and similar fns)

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1631]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1627
- Created at 2012-11-08 15:29:57 by [simo](https://pagure.io/user/simo)
- Closed at 2020-03-24 14:20:41 as wontfix
- Assigned to nobody
---

Although using a hash table is a clever way to diff the lists, it seem like heavy handed both memory and cpu wise.
It occurs to me that a simpler algorithm may be used:
a) copy list1 as list1_only and list2 as list2_only
b) sort list1_only and list2_only
c) create both_list as an empty array of size = list1 + list2
d) walk both lists at the same time comparing values and copying them to both_list at the same time, on a match NULL out both list1_only and list2_only values and add only once to both_list
e) resort list1_only and list2_only pushing NULLs away and keeping count of the remaining values

If the function could take in input list1 and list2 sizes as well instead of having to count them, we could save some cycles there too.

### Comments

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-11-08 15:44:45](https://pagure.io/SSSD/sssd/issue/1627#220617)**

Internal optimization.

milestone: NEEDS_TRIAGE => SSSD 1.10 beta
rhbz:  => 0
summary: Improve efficiency of diff_string_lists (and similar fns) => [RFE] Improve efficiency of diff_string_lists (and similar fns)

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1627#220618)**

Fields changed

selected:  => Not need

---

**Comment from [simo](https://pagure.io/user/simo) at [2013-01-02 15:32:09](https://pagure.io/SSSD/sssd/issue/1627#220619)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-09-26 18:49:57](https://pagure.io/SSSD/sssd/issue/1627#220620)**

Fields changed

mark:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-10 17:06:51](https://pagure.io/SSSD/sssd/issue/1627#220621)**

Fields changed

changelog:  => 
milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: minor => trivial
review:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-12 20:36:09](https://pagure.io/SSSD/sssd/issue/1627#220622)**

Mass-moving tickets not planned for any immediate release and re-setting priority.

milestone: SSSD 1.13 backlog => SSSD Deferred
priority: trivial => major

---

**Comment from [simo](https://pagure.io/user/simo) at [2017-02-24 14:22:16](https://pagure.io/SSSD/sssd/issue/1627#220623)**

**Metadata Update from @simo**:
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [simo](https://pagure.io/user/simo) at [2020-03-24 14:20:40](https://pagure.io/SSSD/sssd/issue/1627#636931)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [simo](https://pagure.io/user/simo) at [2020-03-24 14:20:42](https://pagure.io/SSSD/sssd/issue/1627#636932)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
